File: rithm_algorithms/sorting/insertion_sort.js
let insertionSort = (arr) => {
  for (let i = 0; i < arr.length; i ++) {
    for (let j = arr.length; j > i; j--) {
      if (arr[j] < arr[j - 1]) {
        [arr[j], arr[j -1]] = [arr[j -1], arr[j]]
      }
    }
  }
  return arr;
}

var arr = [5, 3, 2, 4, 1, 5, 3, 4, 2, 3, 3, 4, 5, 8, 7, 77, 66, 5, 4, 32, 3, 4, 5, 6, 77, 6, 5, 4, 3, 2, 2, 2, 1, 9, 8, 7];
console.log(insertionSort(arr));

File: mit_intro_to_algorightms/sorting/insertion_sort.py
def insertion_sort(list):

    def pair_swap(key, last):
        if key == 0 or list[key] > list[last]:
            return

        temp = list[last]
        list[last] = list[key]
        list[key] = temp

        pair_swap(key - 1, last - 1)


    for i in range(1, len(list)):
        if list[i] < list[i - 1]:
            pair_swap(i, i - 1)

    return list
        



print(insertion_sort([5, 2, 4, 6, 1, 3])) # [1, 2, 3, 4, 5, 6]
